Lib/concurrent/futures/thread.py

```diff
@@ -96,8 +98,12 @@ def __init__(self, max_workers=None, thread_name_prefix=''):
         if max_workers <= 0:
             raise ValueError("max_workers must be greater than 0")
 
+        # check that max_queue_size is a positive integer
+        if type(max_queue_size) is not int or max_queue_size < 0:
+            raise ValueError("max_queue_size must be equal or greater 0")
+
         self._max_workers = max_workers
-        self._work_queue = queue.Queue()
+        self._work_queue = queue.Queue(maxsize=max_queue_size)
         self._threads = set()
```

Comment:
`maxsize` is the sole arg to the initializer of `Queue` so you could just supply `max_queue_size` since it's also pretty descriptive.